### PR TITLE
Make dls_release add release tag to repository before build

### PR DESCRIPTION
**Pull request by GDYendell**

### CODE REVIEW

**Comment by ulrikpedersen** [comment]
The default presumeably is (read: should be) to add the tag to the HEAD of the master branch. However, you also need to support use of the -b,--branch flag where the tag is done on the HEAD of a named branch. Does this do that? :-)

File: dls_ade/vcs_git.py [hunk 1]
```diff
@@ -603,16 +603,16 @@ def check_version_exists(self, version):
         Check if version corresponds to a previous release.
 
         Args:
-            version(bool): True or False for whether the version exists or not
+            version(str): True or False for whether the version exists or not
```
Comment (ulrikpedersen) [inline dls_ade/vcs_git.py]: type and description don't seem to match... It's a version-string, right?

File: dls_ade/vcs_git.py [hunk 2]
```diff
@@ -603,16 +603,16 @@ def check_version_exists(self, version):
...
 
         Returns:
-            str: Release tag to check for
+            bool: Release tag to check for
```
Comment (ulrikpedersen) [inline dls_ade/vcs_git.py]: Same as above: type and description don't seem to match... Here I think you're returning a true/false boolean, right?

File: dls_ade/vcs_git.py [hunk 3]
```diff
@@ -627,7 +627,11 @@ def set_version(self, version):
         self._version = version
 
     def release_version(self, version):
-        raise NotImplementedError('version release for git not implemented')
+
+        self.client.create_tag(version, "Release " + version)
+
+        origin = self.client.remotes.origin
+        origin.push(self.client.active_branch, '--follow-tags')
```
Comment (ulrikpedersen) [inline dls_ade/vcs_git.py]: You sure this is right? ASFAIK the git CLI "push" has a "--tags" option, but no "--follow-tags".

**Comment by GDYendell** [comment]
I am testing this now, not ready to be merged.

**Comment by GDYendell** [comment]
Test builds worked for:
a current tag
a new tag
a new tag on a branch
a new tag on a branch with a non-default message

It seems as though it works as expected. Is there anything else you want me to try before merging this @ulrikpedersen?

**Comment by ulrikpedersen** [comment]
Nah, that's all I could think about... Go ahead and merge, @GDYendell 

Thanks! :+1:

**Pull request merged**